source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidRealTimeMultiplayerClient.cs

```diff
@@ -34,111 +54,719 @@ public AndroidRealTimeMultiplayerClient(AndroidJavaObject account)
         {
-            // Task<AnnotatedData<InvitationBuffer>> InvitationsClient.loadInvitations()
+            callback = ToOnGameThread(callback);
+            using (var task = mInvitationsClient.Call<AndroidJavaObject>("loadInvitations"))
+            {
+                AndroidTaskUtils.AddOnSuccessListener<AndroidJavaObject>(
+                    task,
+                    annotatedData =>
+                    {
+                        using (var invitationBuffer = annotatedData.Call<AndroidJavaObject>("get"))
+                        {
+                            int count = invitationBuffer.Call<int>("getCount");
+                            Invitation[] invitations = new Invitation[count];
+                            for (int i = 0; i < count; i++)
+                            {
+                                Invitation invitation;
+                                using (var invitationObject = invitationBuffer.Call<AndroidJavaObject>("get", i))
+                                {
+                                    invitation = AndroidJavaConverter.ToInvitation(invitationObject);
+                                }
+
+                                invitations[i] = invitation;
+                            }
+
+                            callback(invitations);
+                        }
+                    });
+
+                AndroidTaskUtils.AddOnFailureListener(
+                    task,
+                    exception => callback(null));
+            }
+        }
+
+        private static Action<T> ToOnGameThread<T>(Action<T> toConvert)
+        {
+            return (val) => PlayGamesHelperObject.RunOnGameThread(() => toConvert(val));
         }
 
         public void AcceptFromInbox(RealTimeMultiplayerListener listener)
         {
-            // Task<Intent> getInvitationInboxIntent()
+            AndroidHelperFragment.ShowInvitationInboxUI((status, invitation) =>
+            {
+                if (status != UIStatus.Valid)
+                {
+                    OurUtils.Logger.d("User did not complete invitation screen.");
+                    listener.OnRoomConnected(false);
+                    CleanSession();
+                    return;
+                }
+
+                mInvitation = invitation;
+
+                AcceptInvitation(mInvitation.InvitationId, listener);
+            });
         }
 
         public void AcceptInvitation(string invitationId, RealTimeMultiplayerListener listener)
         {
-            // Task<Void> join(@NonNull RoomConfig config)
+            lock (mSessionLock)
+            {
+                int roomStatus = GetRoomStatus();
+                if (roomStatus == ROOM_STATUS_ACTIVE)
+                {
+                    OurUtils.Logger.e("Received attempt to accept invitation without cleaning up " +
+                                      "active session.");
+                    listener.OnRoomConnected(false);
+                    CleanSession();
```

Comment:
We should not break old session in such cases

Reply:
agree, good catch, thanks!